lib/src/rules/prefer_mixin.dart

```diff
@@ -56,9 +58,22 @@ class _Visitor extends SimpleAstVisitor<void> {
   void visitWithClause(WithClause node) {
     for (var type in node.mixinTypes) {
       final element = type.name.staticElement;
-      if (element is ClassElement && !element.isMixin) {
+      if (element is ClassElement &&
+          !element.isMixin &&
+          !isAllowed(element.thisType)) {
         rule.reportLint(type);
       }
     }
   }
+
+  /// Check for "legacy"  classes that cannot easily be made `mixin`s for
+  /// compatibility reasons.
+  /// (See: https://github.com/dart-lang/linter/issues/2082)
+  static bool isAllowed(DartType type) =>
+      DartTypeUtilities.isClass(type, 'IterableMixin', 'dart.core') ||
```

Comment:
For efficiency's sake, consider adding a new utility:
```
static bool isClassElement(ClassElement element, String className, String library) =>
    element.name == className &&
    element.library?.name == library;
```
and using it directly, rather than creating a type just to extract the element from it.

Reply:
Good catch.  Done!